src/main/scala/au/com/agiledigital/toolform/app/ToolFormApp.scala

```diff
@@ -13,38 +20,90 @@ object ToolFormApp extends App {
     * @param args Command line arguments.
     * @return Either[String, Config] - The configuration or an error.
     */
-  def execute(args: Array[String]): Either[String, Config] = {
-    val parser = new scopt.OptionParser[CommandConfig]("toolform") {
+  def execute(args: Array[String]): Either[ToolFormError, String] =
+    for {
+      toolConfiguration <- parseCommandLineArgs(args)
+      project           <- readProject(toolConfiguration.in)
+      summary           <- projectSummary(project)
+    } yield summary
+
+  private def parseCommandLineArgs(args: Array[String]): Either[ToolFormError, ToolFormConfiguration] = {
+    val parser = new scopt.OptionParser[ToolFormConfiguration]("toolform") {
       head("toolform", "0.1")
       help("help").abbr("h").text("Displays this usage text.")
       version("version").abbr("v").text("Displays version information.")
+      opt[File]('i', "inspect") required () valueName "<file>" action { (x, c) =>
+        c.copy(in = x, inspect = true)
+      } text "Displays a summary of the project definition."
     }
 
-    parser.parse(args, CommandConfig()) match {
-      case Some(commandConfig) => Right(ConfigFactory.empty())
-      case None                => Left(s"Invalid arguments - toolform failed.")
-    }
+    parser.parse(args, ToolFormConfiguration()).toRight(ToolFormError(s"Invalid arguments - toolform failed."))
   }
 
-  def displayConfiguration(config: Config): Unit = {
-    val renderOptions: ConfigRenderOptions =
-      ConfigRenderOptions
-        .defaults()
-        .setComments(false)
-        .setOriginComments(false)
-    println(config.root().render(renderOptions))
+  private def readProject(projectDefinitionFile: File): Either[ToolFormError, Project] =
+    if (!projectDefinitionFile.exists()) {
+      Left(ToolFormError(s"File [$projectDefinitionFile] does not exist."))
+    } else if (!projectDefinitionFile.isFile) {
+      Left(ToolFormError(s"File [$projectDefinitionFile] is not a file."))
+    } else {
+      // Read in the root configuration file.
+      Try(ConfigFactory.parseFile(projectDefinitionFile)) match {
+        case Success(simpleConfig) =>
+          // Resolve the configuration aka. replace variable substitutions.
+          val config = simpleConfig.resolve()
+
+          val projectResult = loadConfig[Project](config)
+          resultOrCollectReadErrors(projectResult)
+        case Failure(e) =>
+          Left(ToolFormError(s"Failed to parse project configuration [$projectDefinitionFile]: [${e.getMessage}]"))
+      }
+    }
+
+  private def resultOrCollectReadErrors(projectResult: Either[ConfigReaderFailures, Project]): Either[ToolFormError, Project] =
+    projectResult.left.map(failures => {
+      val failureDetails: String = failures.toList
+        .map(failure => {
+          val failureMessage: String = failure.description + " @ " + failure.location.get.description
```

Comment:
`.get` looks suspicious here. Since `None.get` may fail, and it is up to the developer's reasoning to determine whether it is safe to call in the particular circumstances, we never use it in production code.

Please change to handle the optional location.